data-pipeline/src/telemetry/mod.rs

```diff
@@ -108,13 +111,83 @@ pub struct TelemetryClient {
     handle: JoinHandle<()>,
 }
 
+/// Telemetry describing the sending of a trace payload
+/// It can be produced from a [`SendWithRetryResult`] or from a [`SendDataResult`].
+#[derive(PartialEq, Debug, Default)]
+pub struct SendPayloadTelemetry {
+    requests_count: u64,
+    errors_network: u64,
+    errors_timeout: u64,
+    errors_status_code: u64,
+    bytes_sent: u64,
+    chunks_sent: u64,
+    chunks_dropped: u64,
+    responses_count_per_code: HashMap<u16, u64>,
+}
+
+impl From<&SendDataResult> for SendPayloadTelemetry {
+    fn from(value: &SendDataResult) -> Self {
```

Comment:
nit: I think there is no test for this function. Right now it's just assigning and cloning values but if something changes in `SendDataResult` the test might come in useful.